### PR TITLE
test: Refactor gov e2e for v0.45 Rho

### PR DESCRIPTION
**Pull request by glnro**
This PR fulfills the gov testing requirement for #1909. 

Changes
- Refactored `GovSoftwareUpgrade` and `GovCancelSoftwareUpgrade` to v0.45 SDK cli api
- Refactored `govProposeNewGlobalfee`
- Added `GovCommunityPoolSpend` test

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -288,11 +288,10 @@ replace (
 
 	// force all dependecies to use the same versions of ibc, tendermint and cosmos-sdk
 	github.com/cosmos/ibc-go/v3 => github.com/cosmos/ibc-go/v3 v3.4.0
-	github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.24
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.11
```
Comment (okwme) [inline go.mod:292]: i think we still want to keep this to ensure that all dependencies are using the same version of cosmos sdk (liquidity module etc)

File: go.mod [hunk 2]
```diff
@@ -288,11 +288,10 @@ replace (
 
 	// force all dependecies to use the same versions of ibc, tendermint and cosmos-sdk
 	github.com/cosmos/ibc-go/v3 => github.com/cosmos/ibc-go/v3 v3.4.0
```
Comment (okwme) [inline go.mod:290]:
```suggestion
	github.com/cosmos/ibc-go/v3 => github.com/cosmos/ibc-go/v3 v3.4.0
	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.11
```

File: go.mod [hunk 3]
```diff
@@ -288,11 +288,10 @@ replace (
...
-	github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.24
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.11
 
 	// use cosmos style protobufs
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+	github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.24
```
Comment (okwme) [inline go.mod:295]: why move down here?

**Review by okwme — CHANGES_REQUESTED** [review]
LGTM except for removing the forced SDK compatibility.

File: go.mod (hunk 1 above)
Comment (glnro) [reply go.mod:292]: I think this got messed up when I brought it up to date with main

File: go.mod (hunk 3 above)
Comment (glnro) [reply go.mod:295]: see above comment

**Review by okwme — APPROVED** [review]

**Pull request merged**